public/directives/wz-table/wz-table-directive.js

```diff
@@ -487,23 +492,48 @@ app.directive('wzTable', function() {
       };
 
       $scope.showTooltip = (id1, id2, item) => {
-        const $element = $('#td-' + id1 + '-' + id2 + ' div');
-        const $c = $element
-          .clone()
-          .css({ display: 'inline', width: 'auto', visibility: 'hidden' })
-          .appendTo('body');
-        if (
-          $c.width() > $element.width() &&
-          (($element || [])[0].children || [])[0].innerText !== '-'
-        ) {
+        const $element = $('#td-' + id1 + '-' + id2 + ' div span.wz-text-truncatable');
+        if($element[0].offsetWidth < $element[0].scrollWidth){
           if (!item.showTooltip) {
             item.showTooltip = [];
           }
           item.showTooltip[id2] = true;
         }
-        $c.remove();
       };
 
+      const filterableColumns = () => {
+        $scope.filterableColumns = [];
+        $scope.keys.forEach(k => {
+          const key = $scope.parseKey(k);
+          const canFilter = (($scope.path === '/rules' && (key === 'level' || key === 'pci' || key === 'gdpr' || key === 'groups' || key === 'file' || key === 'path')) ||
+            ($scope.path === '/decoders' && (key === 'path' || key === 'file')))
+          if (canFilter) {
+            $scope.filterableColumns[key] = true;
+          }
+        });
+      };
+
+      $scope.handleClick = (key, item, ev) => {
+        const value = $scope.parseValue(key, item)
+        let keyTmp = $scope.parseKey(key);
+        const valueTmp = typeof value !== 'string' ? value.toString() : value;
+        if ($scope.filterableColumns[keyTmp]) {
+          if (value !== '-' && keyTmp !== 'file') {
+            // only 'group' is accepted as a filter so it has to be overwritten
+            if (keyTmp === 'groups')
+              keyTmp = 'group';
+            // PCI, GDPR and GROUPS can contains multiple values and the api only supports one at a time, just the first value is sent
```

Comment:
This doesn't make sense, instead, we should get the exactly clicked filter. If not applicable, we need a ticket opened in the Wazuh API and remove this code from here.

Reply:
It can make sense. When we ask for the rules, they can belong to one or several groups, so the object we receive is an array called groups:

```js
            "groups": [ 
               "syslog 
            ],
```
On the other hand, when we filter and do a search for rules, for groups it is only allowed to filter by some, so the filter is called `group`.

Then, if we click on the Key groups, we take the first group and we filter through it with the `group` parameter.